### PR TITLE
Delegate custom deletes

### PR DESCRIPTION
**Pull request by abnegate**
## What does this PR do?

Move database and collections deletes to the database worker

## Test Plan

Manual tests

## Checklist

- [x] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [x] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

**Review by eldadfux — APPROVED** [review]

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -3532,8 +3532,10 @@ function updateAttribute(
 
         $processDocument($collection, $document);
 
-        $queueForDeletes
-            ->setType(DELETE_TYPE_AUDIT)
+        $queueForDatabase
+            ->setType(DATABASE_TYPE_DELETE_DOCUMENT)
+            ->setDatabase($database)
+            ->setCollection($collection)
```
Comment (christyjacob4) [inline app/controllers/api/databases.php:3538]: This is the delete document endpoint.. Why do we need to send this to a worker? We delete the document in the API itself right?

File: src/Appwrite/Platform/Workers/Databases.php [hunk 2]
```diff
@@ -480,27 +455,179 @@ private function deleteIndex(Document $database, Document $collection, Document
...
+     * @param Document $project
+     * @param Database $dbForProject
+     * @return void
+     * @throws Authorization
+     * @throws Conflict
+     * @throws DatabaseException
+     * @throws Restricted
+     * @throws Structure
+     */
+    protected function deleteCollection(Document $database, Document $collection, Document $project, Database $dbForProject): void
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Databases.php:494]: Shouldn't we now remove this code from the deletes worker ?

File: src/Appwrite/Platform/Workers/Databases.php [hunk 3]
```diff
@@ -480,27 +455,179 @@ private function deleteIndex(Document $database, Document $collection, Document
...
     }
+
+    /**
+     * @param Document $database
+     * @param Document $project
+     * @param $dbForProject
+     * @return void
+     * @throws Exception
+     */
+    protected function deleteDatabase(Document $database, Document $project, $dbForProject): void
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Databases.php:471]: Shouldn't we now remove this code from the deletes worker ?

File: app/controllers/api/databases.php (hunk 1 above)
Comment (eldadfux) [reply app/controllers/api/databases.php:3538]: I think this is for delete the document audit and not the document itself. We can leave it as is.

File: src/Appwrite/Platform/Workers/Databases.php (hunk 2 above)
Comment (eldadfux) [reply src/Appwrite/Platform/Workers/Databases.php:494]: I think we can.

File: src/Appwrite/Platform/Workers/Databases.php (hunk 3 above)
Comment (eldadfux) [reply src/Appwrite/Platform/Workers/Databases.php:471]: I think we can.

**Pull request merged**